ucsb-cs156-s24/proj-happycows-s24-4pm-8
# JM - 4F - Button and Modal to Remove and Restore Users (Merge after B3, 24)

## Overview
This PR builds off of `4-B3-JeffreyRemoveUsers` and touches 5 different files: `usersFixtures.js`, `UsersTable.js`, `users.js`, and their relevant test files. This PR also address both issue #26 -- Button to suspend/restore users -- and issue #27  -- Confirmation modal to suspend/restore users -- in the same PR.

**NOTE**: The prior PR for `4-B3-JeffreyRemoveUsers` and the PR for `24-JeffreyIncrementalPitest` must be merged before touching this one.

Link to dokku deployment is below in the Validation section.
## Screenshots (Optional)
### Workflow to Suspend
<img width="1335" alt="image" src="https://github.com/ucsb-cs156-s24/proj-happycows-s24-4pm-8/assets/61306390/a42388cf-0d30-4970-8d21-fc125cb96e51">

<img width="1124" alt="image" src="https://github.com/ucsb-cs156-s24/proj-happycows-s24-4pm-8/assets/61306390/f4c6628e-b00e-4665-b66d-a1c8f601ea19">

<img width="931" alt="image" src="https://github.com/ucsb-cs156-s24/proj-happycows-s24-4pm-8/assets/61306390/90f7d527-ebda-4e8b-863e-24b29d53c83b">

### Workflow to Restore
<img width="931" alt="image" src="https://github.com/ucsb-cs156-s24/proj-happycows-s24-4pm-8/assets/61306390/90f7d527-ebda-4e8b-863e-24b29d53c83b">

<img width="1141" alt="image" src="https://github.com/ucsb-cs156-s24/proj-happycows-s24-4pm-8/assets/61306390/8b9c9cdc-903f-4c6f-99ea-5fad300fd36d">

<img width="923" alt="image" src="https://github.com/ucsb-cs156-s24/proj-happycows-s24-4pm-8/assets/61306390/06aac2c0-4c96-4a05-9494-d59e759a6f53">

## Feedback Request (Optional)
- I colored the `Restore` and `Suspend` buttons differently to differentiate them. Is this a welcome coloring?
- Additionally, I chose to use the user's email rather than their name for identification purposes because that's been the standard for the backend as well (user can only be queried by email).

## Validation
1. Navigate to the dokku deployment: https://happycows-qa.dokku-08.cs.ucsb.edu 
2. Log In
3. Go to this page -- https://happycows-qa.dokku-08.cs.ucsb.edu/admin/users
4. Test this function.
5. Refresh to see changes

## Tests

## Linked Issues

Closes #26 
Closes #27
